code/modules/mob/living/simple_animal/hostile/hostile2.dm

```diff
@@ -169,6 +169,7 @@
 	LostTarget()
 
 /mob/living/simple_animal/hostile/proc/Goto(var/target, var/delay, var/minimum_distance)
+	step_towards(src, target) //weird but necessary so they try to bump openable obstacles
```

Comment:
I'll have to take a more in depth look tomorrow when I have more time, but right now it seems odd to be calling both walk_to and step_towards. I feel like there's a lot of issues that could come out of calling both. For instance, would this not override the use of minimum_distance below?  If a mob has a minimum_distance set, won't they just get stuck walking back and forth between two tiles to maintain their minimum distance? Have you tested movement much with this?